tsdb/isolation.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+import (
+	"sync"
+)
+
+// IsolationState holds the isolation information.
```

Comment:
Is there a point in making this public? That should probably be internal.

Reply:
I was asking myself the same question, but then thought that @brian-brazil and/or @gouthamve probably had a reason. And then I stopped thinking about it… (o: But you are right, I guess it should be unexported.

Reply:
This does not seem to be usable with public apis in any ways...

Reply:
Unexported IsolationState.